tests/unit/s2n_ems_extension_test.c

```diff
@@ -67,5 +67,45 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_connection_free(client_conn));
     }
 
+    /* s2n_server_ems_is_missing */
+    {
+        struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT);
+        EXPECT_NOT_NULL(conn);
+
+        /**
+         *= https://tools.ietf.org/rfc/rfc7627#section-5.3
+         *= type=test
+         *#    If the original session used the extension but the new ServerHello
+         *#    does not contain the extension, the client MUST abort the
+         *#    handshake.
+         **/
+        conn->ems_negotiated = true;
+        EXPECT_FAILURE_WITH_ERRNO(s2n_server_ems_extension.if_missing(conn), S2N_ERR_MISSING_EXTENSION);
+
+        conn->ems_negotiated = false;
+        EXPECT_SUCCESS(s2n_server_ems_extension.if_missing(conn));
+
+        EXPECT_SUCCESS(s2n_connection_free(conn));
+    }
+
+    /* s2n_client_ems_should_send */
+    {
+        struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT);
+        EXPECT_NOT_NULL(conn);
+
+        EXPECT_TRUE(s2n_client_ems_extension.should_send(conn));
```

Comment:
nit: geral comment about testing. because of the setup required, it sometime hard to konw what exactly we want to be testing. might i suggest something like:

```
/* Setup
 * - create the connection
 * - verify connection should send ems extension
*/
... code

/* Setup
 * - attach extension with ems_negotiated to connection
*/
.. code

/* Expectation
 * - verify connection should send ems extension
*/
.. code
```

Reply:
I am testing three different cases inside of the s2n_client_ems_should_send unit test. I will add comments to make each case more clear.